File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.java
class Solution {
    public int findMaxK(int[] nums) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        int largest=Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++)
        {
            if(mp.containsKey(-1*nums[i]))
            {
               largest=Math.max(largest,Math.abs(nums[i]));
            }
            mp.put(nums[i],i);
        }
        if(largest==Integer.MIN_VALUE) return -1;
        else return largest; 
    }
}

File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
        set<int> st; int maxi=-1; 
        for(int i=0;i<nums.size();i++)
        {
            st.insert(nums[i]);
            if(st.find(-1*nums[i])!=st.end())
            {
                maxi=max(maxi,abs(nums[i]));
            }
        }
        return maxi;
    }
};